api/v1/controllers/collectors.js

```diff
@@ -285,6 +283,24 @@ function heartbeat(req, res, next) {
     return o.save();
   })
 
+  // get the ids for changed generators
+  .then(() => heartbeatUtils.getChanges(collectorName))
+
+  // find the generator objects for the changes ids
+  .then((changed) => Promise.join(
+      Generator.findAll({ where: { id: changed.added } }),
+      Generator.findAll({ where: { id: changed.deleted } }),
+      Generator.findAll({ where: { id: changed.updated } }),
```

Comment:
Is changed.[added/updated/deleted] a list of ids?

Reply:
Yes. I will change it to make it more clear.